### DISCUSSION TITLE
[API] DELETE object in bucket - Returns object not found

### DISCUSSION THREAD

**Discussion by olivierb24** [supabase/supabase#19518] [Questions] [answered]
I've been trying for a few days to delete files from a bucket via API with no luck. Tried adding RLS policies(set to TRUE)  removing them, making buckets public or private. Nothing seems to work.

The swagger API docs show that it's possible to delete at this endpoint:

{supabaseurl}/storage/v1/object/bucket/filePath.

Using 'DELETE' and secret_role key as 'authorization' header. 

I can't seem to figure out how to delete. Other calls like signing a file URL work fine. Anyone else had success with the storage API?

**Comment by GaryAustin1** [comment]
Can you show an example filepath?  If you have your apikey and authorization header to service_role then RLS won't matter.

**Comment by olivierb24** [comment, marked as answer]
Hi,

thanks for replying so quickly! I've actually got it working now. My
mistake.

apiKey was not set properly to service_role.

Thanks for replying!! Take care,

Olivier
